addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -95,9 +95,9 @@ export class ProductScreen extends Component {
             : this.pos.models["pos.category"].filter((category) => !category.parent_id);
     }
     getCategoriesAndSub() {
-        return this.getAncestorsAndCurrent().flatMap((category) =>
-            this.getChildCategoriesInfo(category)
-        );
+        return this.getAncestorsAndCurrent()
+            .flatMap((category) => this.getChildCategoriesInfo(category))
+            .sort((a, b) => a.id - b.id);
```

Comment:
```suggestion
            .toSorted((a, b) => a.id - b.id);
```
technically mutating with `sort` shouldn't cause any problems, but if you can avoid it why not

Reply:
first time i see this method, thank you.